### ISSUE TITLE
Incorporate Flame's Audio functionalities

### ISSUE THREAD

**Issue by luanpotter** [luanpotter/audioplayer#18]
Flame's [audio class](https://github.com/luanpotter/flame/blob/master/lib/audio.dart) allows for some neat methods for playing resource audio files. It'd be very useful to actually import that behavior to audioplayers itself, and just use it on flame, because this is really useful to all applications (not just games).

The relevant code is this:

```dart
  Future<File> _fetchToMemory(String fileName) async {
    final file = new File('${(await getTemporaryDirectory()).path}/$fileName');
    return await file
        .writeAsBytes((await _fetchAsset(fileName)).buffer.asUint8List());
  }
```

It's loading a resource into a tmp folder in the device file system, which is then used with the isLocal flag to play.

Basically we should have 3 kinds of sources:

* local file on the user device
* resource file
* remote url (over network)

And it'd be good to make it more clear in the documentation what kinds are available and what they mean.

Also, maybe we could support in memory streams? Just a thought.

**Comment by luanpotter** [comment]
This was added on 0.6.0.
